EIPS/eip-4362.md

```diff
@@ -0,0 +1,279 @@
+  - Let `namespace` be the `namespace` in the CapGrok URI entry and let `capDetails` be the base64url-decoded CapGrok Details Object of the CapGrok URI entry.
+  - Let `defaultActions` be the `def` JSON array in `capDetails`, where each value represents an action.
+  - If `defaultActions` is present, do the following:
+    - Let `actions` be the string concatenation of each action in the array with the delimiter `", "`.
+    - Append the string concatenation of `" ("`, `numbering`, `")"` to `capgrok-transformed-statement`.
+    - Append `namespace` concatenated with `": "` to `capgrok-transformed-statement`.
+    - Append `actions` to `capgrok-transformed-statement`.
+    - Append the string `" for any."` to `capgrok-transformed-statement`.
+    - Increase `numbering` by 1.
+  - Let `targetedActions` be the `tar` JSON object in `capDetails`, where each key-value pair represents the set of actions allowed for a target.
+  - If `targetedActions` is present, do the following:
+    - Let `actionSets` be an array of arrays of strings;
+    - For each key-value pair in `targetedActions`, ordered alphabetically by key, append the string array value to `actionSets`.
+    - For each array of strings `actionSet` in `actionSets`, do the following:
+      - Sort the strings in `actionSet` alphabetically.
+      - Let `actions` be the string concatenation of each action in the array with the delimiter `", "`.
+      - Let `targets` be the string concatenation of each key in `targetedActions` with the delimiter `", "`, for those keys such that the associated value (or any permutation of that value) is identical to `actionSet`.
+      - Append the string concatenation of `" ("`, `numbering`, `")"` to `capgrok-transformed-statement`.
+      - Append `namespace` concatenated with `": "` to `capgrok-transformed-statement`.
+      - Append `actions` to `capgrok-transformed-statement`.
+      - Append the string `" for "` to `capgrok-transformed-statement`.
+      - Append `targets` to `capgrok-transformed-statement`.
+      - Append the string `" ."` to `capgrok-transformed-statement`.
+      - Increase `numbering` by 1.
+- Return `capgrok-transformed-statement`.
+
+#### CapGrok Verification
+
+The following algorithm or an algorithm that produces the same output MUST be performed to verify a SIWE CapGrok.
+
+Inputs:
+- Let `capgrok-siwe` be the input SIWE message conforming to EIP-4361 and this EIP.
+Algorithm:
+- Let `uri` be the uri field of `capgrok-siwe`.
+- Let `capgrok-uris` be an array of CapGrok URIs taken in order from the resources field of `capgrok-siwe`, such that URIs which are not valid CapGrok URIs are ignored.
+- Let `capgrok-transformed-statement` be the result of performing the above `CapGrok Translation Algorithm` with `uri` and `capgrok-uris` as input.
+- Assert that the statement field of `capgrok-siwe` ends with `capgrok-transformed-statement`.
+
+### Implementer's Guide
+
+(non-normative)
```

Comment:
```suggestion

TBD
```
Required: Remove that non-normative thing. It's part of the spec, so it is normative..